test/distributed/_composable/test_compose.py

```diff
@@ -171,6 +179,37 @@ def test_checkpoint_fsdp_submodules_with_param_no_shard(self):
             self._test_parity,
         )
 
+    def test_composable_fsdp_replicate(self):
+        model = CompositeModel(device=torch.device("cpu"))
+        fully_shard(model.l1)
+        with self.assertRaisesRegex(
+            AssertionError, "Cannot apply .*replicate"
+        ):
+            replicate(model.l1)
+        replicate(model.l2)  # should not raise
```

Comment:
would it be useful to run a forward and backward pass to validate?

Reply:
Thanks!
This test only verifies how the APIs can be composed. For example, if apply both `fully_shard` and `replicate` on the same module, it should raise exception. Let me add a comment here.
The correctness of `forward` and `backward` are verified in the following test.